src/cursor/cursor.ts

```diff
@@ -114,20 +125,17 @@ export interface CursorOptions extends CoreCursorOptions {
  *
  * collection.find({}).maxTimeMS(1000).maxScan(100).skip(1).toArray(..)
  */
-export class Cursor extends CoreCursor {
-  /**
-   * @param {any} topology
-   * @param {any} ns
-   * @param {any} [cmd]
-   * @param {any} [options]
-   */
-  s: any;
-  constructor(topology: any, ns: any, cmd?: any, options?: any) {
+export class Cursor<
+  T extends CursorOptions = CursorOptions,
+  O extends OperationBase = OperationBase
+> extends CoreCursor<T, O> {
+  s: CursorPrivate;
+
+  constructor(topology: Topology, ns: string | O, cmd: Document, options: T = {} as T) {
     super(topology, ns, cmd, options);
 
-    options = options || {};
     if (this.operation) {
-      options = this.operation.options;
+      options = this.operation.options as T;
```

Comment:
I think if you remove the casting, this should actually catch errors where the options are not the same as the operation.

Reply:
I get this error:
```
Type 'OperationOptions' is not assignable to type 'T'.
  'OperationOptions' is assignable to the constraint of type 'T', but 'T' could be instantiated with a different subtype of constraint 'CursorOptions'
```
I don't think it can catch the type errors because none exist technically, but they could if the subtype has properties that aren't defined on the base type.